dhis-2/dhis-web-api/src/test/java/org/hisp/dhis/webapi/controller/SpringBindingTest.java

```diff
@@ -103,7 +104,7 @@ void shouldReturnABadRequestWhenInvalidValueForIntegerInBindingObjectIsPassedAsP
             .param( "integerNumber", "10.5" ) )
             .andExpect( content().string( containsString( "Bad Request" ) ) )
             .andExpect( content().string( containsString(
-                "Value 10.5 is not valid for parameter integerNumber. It should be of type Integer" ) ) );
+                "Value 10.5 is not valid for parameter integerNumber. For input string: \"10.5\"" ) ) );
```

Comment:
This suffix seems redundant to the beginning of the message 🤔 It would not help me as a user to know why `10.5` is not valid and what I should do about it. Where does this one come from?

Reply:
Maybe this needs some agreement on the types that we use as parameters.
The part of the message `It should be of type X` would make sense for some types but not for others that are just implementation details.
As now, we are just checking if the type is primitive and in that case telling the user to use that type, in all the other cases we are returning the message generated by Spring